### DISCUSSION TITLE
How do you stage and commit "noise" on you forms/reports?

### DISCUSSION THREAD

**Discussion by monheimx9** [joyfullservice/msaccess-vcs-addin#429] [Q&A] [answered]
It's something I've tried to find a solution for myself, but without much success.
What are your processes for increasing staging speed at committing noise on forms/reports?
(I'm using the sanitize option at level Aggressive btw)

I regularly find myself making small changes to the code of certain forms, without affecting their visual appearance, and on export, Access being what it is, I end up with the usual noise on certain controls.

But from time to time I find myself with 20 or 30 forms where I first stage the noise in a single commit, and then the 'effective' modifications.

But the noise process is relatively repetitive and time-consuming, and I haven't come up with any miraculous idea that would allow me to automate this task to some extent

How do you go about it? Do you commit the noise every time you export? Do you wait and do it once every x commits? Do you have a script that lets you do it with a single click?
I'd be very curious to read about your work processes.

Best regards,
Jeremy

**Comment by hecon5** [comment, marked as answer]
With V4 of the Addin, the noise from commits is almost negligible, honestly. We've also spent a lot of time enforcing "spelling" on items, which has made a big improvement on noise, as well. 

For "noise" where I'm confident it's not something driving an actual change, I ignore it and never commit noise to the repository.

**Reply by mwolfe02** [reply]
To piggyback on @hecon5's "spelling" comment, we adhere to a strict capitalization convention to account for VBA's case-changing "feature": [PascalCase All the Things](https://nolongerset.com/pascalcase-all-the-things/).  This helps quite a bit to reduce noise around case changes of identifiers.

**Reply by joyfullservice** [reply]
@mwolfe02 - Yes, that's what we do as well, and almost never have issues with case changes. PascalCase is definitely a big help in VBA.  👍

**Reply by monheimx9** [reply]
Noted, that sounds like a good way to go

There's not much noise to speak of per "object".
In fact, I think it's mostly a problem in my workflow
As some of the features I implement require modifications to many different objects, I end up with noise on all the objects in question.

I'm going to follow what you say and simply not commit this noise, and re-sync the repository after work

**Comment by joyfullservice** [comment]
The main place I see "noise" is when doing development on computers that have different monitor setups or screen resolutions. This impacts some of the (non-critical) screen position values. I usually just ignore those till I am making an actual change on the object, then go ahead and include those with the commit.

The other place I see a lot of "noise" is when changes to the VCS add-in affect how the source files are produced. (Such as filtering out additional unnecessary elements, or formatting changes.) I typically just commit those separately from actual development changes and title them "VCS Updates". It's a pretty simple approach, but seems to work well for us.

**Reply by monheimx9** [reply]
Ah yes indeed, I haven't thought about the impact of multi-monitors setup in the process

Thanks to everybody for taking the time to answer, and also for your commitment to this awesome project
